### PR TITLE
Fix rendering of deprecated annotation in mixins

### LINKED ISSUE #1121: @deprecated on mixin member generates invalid code
Deprecating mixin member causes the code generation to output invalid code. 
Tested with 0.17.12

Given:
```smithy
@mixin
structure Extra {
    @deprecated
    @required
    name: String
}

structure Greeting with [Extra] {
    @required
    message: String
}
```

actual generated code:
```scala
trait Extra {
  def @deprecated(message = "N/A", since = "N/A") name: String
}
```

expected generated code:
```scala
trait Extra {
  @deprecated(message = "N/A", since = "N/A") 
  def name: String
}
```

Full code to reproduce the issue can be found here https://github.com/ghostbuster91/demos/tree/deprecated-issue-mixin

### PR DESCRIPTION
**Pull request by ghostbuster91**
This fixes #1121

### CODE REVIEW

**Review by kubukoz — COMMENTED** [review]
can you make this target 0.17? It'd be great to add in a patch rather than wait for 0.18.

**Review by kubukoz — APPROVED** [review]

**Comment by ghostbuster91** [comment]
> can you make this target 0.17? It'd be great to add in a patch rather than wait for 0.18.

Good idea, changed.

**Comment by daddykotex** [comment]
Thank you so much

**Comment by daddykotex** [comment]
I'll happily port it to 0.18

File: modules/example/src/smithy4s/example/DeprecatedMixin.scala [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package smithy4s.example
+
+
+@deprecated(message = "A compelling reason", since = "0.0.1")
+trait DeprecatedMixin {
+  @deprecated(message = "N/A", since = "N/A") def strings: Option[List[String]]
```
Comment (daddykotex) [inline modules/example/src/smithy4s/example/DeprecatedMixin.scala]: should we add a new line between the annotation and the def?

Comment (daddykotex) [reply modules/example/src/smithy4s/example/DeprecatedMixin.scala]: added it, hope you don't mind

**Review by daddykotex — APPROVED** [review]

**Review by kubukoz — APPROVED** [review]

**Comment by kubukoz** [comment]
@daddykotex I'll open a backport, there's another PR to sync as well so I'll deal with them both

**Comment by daddykotex** [comment]
ahh thanks :)

**Pull request merged**